src/main/kotlin/Main.kt

```diff
@@ -1,3 +1,80 @@
+import java.util.Scanner
+
+abstract class MenuItem {
+    abstract val index: Int
+    abstract val name: String
+}
+class Archive(override val index: Int, override val name: String, val notes: MutableList<Note>?): MenuItem()
+
+class Note(override val index: Int, override val name: String, var content: String): MenuItem()
+
 fun main(args: Array<String>) {
-    println("Hello World!")
-}
\ No newline at end of file
+
+    // пустой список архивов
+    var archives: MutableList<Archive> = mutableListOf()
```

Comment:
```suggestion
    val archives: MutableList<Archive> = mutableListOf()
```
Список и так мьютабельный, делать ссылку мьютабельной не стоит

Reply:
Спасибо, не заметила - поправила